launch_openpilot.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/bash
+
+function launch {
+  # apply update
+  if [ "$(git rev-parse HEAD)" != "$(git rev-parse @{u})" ]; then
+     git reset --hard @{u} &&
+     git clean -xdf &&
+     exec "${BASH_SOURCE[0]}"
+  fi
+
+  # check if NEOS update is required
+  while [ "$(cat /VERSION)" -lt 4 ] && [ ! -e /data/media/0/noupdate ]; do
+    curl -o /tmp/updater https://neos.comma.ai/updater && chmod +x /tmp/updater && /tmp/updater
+    sleep 10
+  done
+
+  # no cpu rationing for now
+  echo 0-3 > /dev/cpuset/background/cpus
+  echo 0-3 > /dev/cpuset/system-background/cpus
+  echo 0-3 > /dev/cpuset/foreground/boost/cpus
+  echo 0-3 > /dev/cpuset/foreground/cpus
+  echo 0-3 > /dev/cpuset/android/cpus
+
+  # wait for network
+  (cd selfdrive/ui/spinner && exec ./spinner 'waiting for network...') & spin_pid=$!
+  until ping -W 1 -c 1 8.8.8.8; do sleep 1; done
```

Comment:
Does this mean that is not going to start until it connects to the internet??

I'm noticing that the spinner is taking longer than normal. And I had to reboot once because it got stuck in the spinner loading screen for more than 5min

Reply:
NEOS never boots to baseui without a successful network ping, in stock form. You can remove this check, but will risk losing the auto-update functionality on reboot.

Reply:
Is a shame that no one from commaai addressed my question. I've been in situations where I have to reset my EON where there is no cellphone coverage, and it's impossible because of this.